homeassistant/components/group/__init__.py

```diff
@@ -134,15 +140,25 @@ def on_off_states(self, on_states: set, off_state: str) -> None:
         self.on_states_by_domain[current_domain.get()] = set(on_states)
 
 
+class GroupProtocol(Protocol):
+    """Define the format of group platforms."""
+
+    def async_describe_on_off_states(
+        self, hass: HomeAssistant, registry: GroupIntegrationRegistry
+    ) -> None:
+        """Describe group on off states."""
+
+
```

Comment:
This does not seem like a good a good place here. Maybe move it up a bit more above `def _conf_preprocess`?